src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1568,6 +1576,21 @@ private function registerMessengerConfiguration(array $config, ContainerBuilder
                 ->setArguments(array($transport['dsn'], $transport['options']))
                 ->addTag('messenger.receiver', array('alias' => $name))
             ;
+
+            if (0 === strpos($transport['dsn'], 'symfony://kernel.terminate')) {
+                $kernelTerminateBusId = $transport['options']['bus'] ?? null;
+                if ($kernelTerminateBusId && !isset($buses[$kernelTerminateBusId])) {
+                    throw new InvalidArgumentException(sprintf('No bus with id "%s" was found in framework.messenger.buses config for transport "%s". Known buses are %s.', $kernelTerminateBusId, $name, json_encode(array_keys($buses))));
+                }
+
+                if (!$kernelTerminateBusId && 1 === \count($buses)) {
```

Comment:
In case `count !== 2` and `!$kernelTerminateBusId` we should throw and exception too, right?

Reply:
Why `count !== 2` 🤔 
We only set the bus from the extension if it wasn't provided and there is only one available.
If there is more than one, the bus might have been set using the `bus` query param in the DSN and will be selected by the factory :)

Reply:
Sorry, I meant !== 1, so if it’s not set by params or dsn, and there is more than one bus, then throw exception

Reply:
Which is handled in the factory after checking it's not in the DSN: https://github.com/symfony/symfony/pull/28746/files#diff-b2491fcc861d34cf32d0b34ad1828c99R37